test/spanner.ts

```diff
@@ -1047,6 +1199,18 @@ describe('Spanner with mock server', () => {
           .catch(done)
           .then(() => done());
       });
+      const gotStreamingCalls = xGoogReqIDInterceptor.getStreamingCalls();
```

Comment:
I added this verification to check that the headers are actually being sent and received. Can you add more of these to the various tests here, because:
1. Currently, the attempt number is never increased. If we want to do that in a follow-up PR, then that is fine, but we should then add a TODO to these verifications.
2. The current 'global' verification for the request headers has a limitation, and that is that it does not seem to catch missing headers for streaming RPCs. It only verifies that if the header is set, that it has the correct format.

Reply:
I've filed a TODO to send a follow-up PR.